### PR TITLE
feat(default_ad_api): add door api

### PR DESCRIPTION
**Pull request by isamu-takagi**
## Description

Add default implementation for [door API](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/ad-api/features/vehicle-doors/) and door simulation tools.
- Door API (default_ad_api)
- Door simulator (vehicle_device_simulator)
- Door panel (tier4_adapi_rviz_plugin)

## Related links

https://github.com/autowarefoundation/autoware_launch/pull/921
https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/ad-api/features/vehicle-doors/

## Tests performed

1. Checkout https://github.com/autowarefoundation/autoware_launch/pull/921 and build.
2. Launch planning simulator.
3. Add door panel on rviz (tier4_adapi_rviz_plugin -> DoorPanel)
4. Press open/close button and check status.

Screenshot from 2023-12-01 13-50-09

## Notes for reviewers

None

## Interface changes

Add
- `/api/vehicles/doors/command`
- `/api/vehicles/doors/layout`
- `/api/vehicles/doors/status`
- `/vehicles/doors/command`
- `/vehicles/doors/layout`
- `/vehicles/doors/status`

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by yabuta — APPROVED** [review]
LGTM

**Review by kosuke55 — APPROVED** [review]

**Review by mitsudome-r — APPROVED** [review]

File: common/component_interface_specs/include/component_interface_specs/vehicle.hpp [hunk 1]
```diff
@@ -71,6 +74,27 @@ struct EnergyStatus
...
+struct DoorLayout
+{
+  using Service = autoware_adapi_v1_msgs::srv::GetDoorLayout;
+  static constexpr char name[] = "/vehicle/doors/layout";
+};
+
+struct DoorStatus
+{
+  using Message = autoware_adapi_v1_msgs::msg::DoorStatusArray;
+  static constexpr char name[] = "/vehicle/doors/status";
```
Comment (mitsudome-r) [inline common/component_interface_specs/include/component_interface_specs/vehicle.hpp:92]: Should we make the names consistent with other command/status messages?

* /vehicle/command/doors_command
* /vehicle/status/doors_status
* /vehicle/doors_layout

or should we change the current message for control commands to something like 
`/vehicle/control/actuation_command` ?

Comment (mitsudome-r) [reply common/component_interface_specs/include/component_interface_specs/vehicle.hpp:92]: @yukkysaito What do you think?

Comment (isamu-takagi) [reply common/component_interface_specs/include/component_interface_specs/vehicle.hpp:92]: @mitsudome-r @yukkysaito 
When thinking of it as a vehicle interface, it feels strange to have separate namespaces for commands and status into `/vehicle` and `/control`. So I propose to change the vehicle interface as follows. 

| current topic name                     | new topic name                          |
| -------------------------------------- | --------------------------------------- |
| /vehicle/status/control_mode           | /vehicle/control/control_mode           |
| /vehicle/status/gear_status            | /vehicle/control/gear_status            |
| /vehicle/status/turn_indicators_status | /vehicle/control/turn_indicators_status |
| /vehicle/status/velocity_status        | /vehicle/control/velocity_status        |
| /vehicle/status/steering_status        | /vehicle/control/steering_status        |
| /control/command/control_cmd           | /vehicle/control/control_cmd            |
| /control/command/gear_cmd              | /vehicle/control/gear_cmd               |
| /control/command/turn_indicators_cmd   | /vehicle/control/turn_indicators_cmd    |
| -                                      | /vehicle/doors/layout                   |
| -                                      | /vehicle/doors/status                   |
| -                                      | /vehicle/doors/command                  |

**Review by yukkysaito — APPROVED** [review]

**Review by taikitanaka3 — APPROVED** [review]

**Pull request merged**